Examples/ObjectiveC/UserTrackingModesExample.m

```diff
@@ -3,23 +3,182 @@
+            arrowFillColor = [[UIColor clearColor] CGColor];
+            arrowRotation = 0;
+            break;
+        case MGLUserTrackingModeFollow:
+            arrowStrokeColor = self.tintColor.CGColor;
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2 + 2, UserLocationButtonSize / 2 - 2);
+            arrowFillColor = [[UIColor clearColor] CGColor];
+            arrowRotation = 0.66;
+            break;
+        case MGLUserTrackingModeFollowWithHeading:
+            arrowStrokeColor = [[UIColor clearColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2 + 2, UserLocationButtonSize / 2 - 2);
+            arrowFillColor = self.tintColor.CGColor;
+            arrowRotation = 0.66;
+            break;
+        case MGLUserTrackingModeFollowWithCourse:
+            arrowStrokeColor = [[UIColor clearColor] CGColor];
+            arrowPosition = CGPointMake(UserLocationButtonSize / 2, UserLocationButtonSize / 2);
+            arrowFillColor = self.tintColor.CGColor;
+            arrowRotation = 0;
+            break;
+    }
+    
+    [_arrow setFillColor:arrowFillColor];
+    [_arrow setStrokeColor:arrowStrokeColor];
+    [_arrow setAffineTransform:CGAffineTransformMakeRotation(arrowRotation)];
+    
+    [self layoutIfNeeded];
+}
+
+@end
+
+@interface UserTrackingModesExample () <MGLMapViewDelegate>
+
+@property (nonatomic) MGLMapView *mapView;
+@property (nonatomic) UserLocationButton IBOutlet *button;
+
+@end
+
 @implementation UserTrackingModesExample
 
 - (void)viewDidLoad {
     [super viewDidLoad];
 
-    MGLMapView *mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle darkStyleURL]];
-    mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView = [[MGLMapView alloc] initWithFrame:self.view.bounds styleURL:[MGLStyle darkStyleURL]];
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView.delegate = self;
 
-    mapView.userTrackingMode = MGLUserTrackingModeFollowWithHeading;
+    self.mapView.tintColor = [UIColor redColor];
+    self.mapView.attributionButton.tintColor = [UIColor lightGrayColor];
 
-    // The user location annotation takes its color from the map view tint.
-    mapView.tintColor = [UIColor redColor];
+    [self.view addSubview:self.mapView];
+    
+    [self setupLocationButton];
+}
+
+- (void)mapView:(MGLMapView *)mapView didChangeUserTrackingMode:(MGLUserTrackingMode)mode animated:(BOOL)animated {
+    [self.button updateArrow:mode];
+}
 
-    // You can set the attribution button tint separately.
-    mapView.attributionButton.tintColor = [UIColor lightGrayColor];
+-(void)locationButtonTapped:(UserLocationButton *)sender {
+    MGLUserTrackingMode mode;
 
-    [self.view addSubview:mapView];
+    switch (_mapView.userTrackingMode) {
```

Comment:
It doesn’t make a functional difference, but let’s be consistent and not use the underscore `self` shorthand anywhere.